src/bootstrap/src/core/build_steps/dist.rs

```diff
@@ -2586,3 +2586,30 @@ impl Step for Gcc {
         tarball.generate()
     }
 }
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct Enzyme {
+    pub target: TargetSelection,
+}
+
+impl Step for Enzyme {
+    type Output = GeneratedTarball;
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.alias("enzyme")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(Enzyme { target: run.target });
+    }
+
+    fn run(self, builder: &Builder<'_>) -> Self::Output {
+        let mut tarball = Tarball::new(builder, "enzyme", &self.target.triple);
+        let output = builder.ensure(super::llvm::Enzyme { target: self.target });
+        tarball.add_dir(output.join("include"), "include");
+        tarball.add_file(output.join("lib").join("libEnzyme-20.so"), ".", FileType::NativeLibrary);
```

Comment:
This probably shouldn't be hardcoded. Either we should reuse [this logic](https://github.com/rust-lang/rust/blob/22280337f632a8f04b16a9d1edb40dac5b542513/src/bootstrap/src/core/build_steps/compile.rs#L1195) or the output of the `llvm::Enzyme` step should contain the filename of the built library (I would prefer that).